### PR TITLE
Spec IS bulk_lookup endpoint

### PR DESCRIPTION
**Pull request by dbkr**
Merge @maxidor 's commits from https://github.com/matrix-org/matrix-doc/pull/1040 (excluding https://github.com/matrix-org/matrix-doc/pull/1039 which I'm merging in https://github.com/matrix-org/matrix-doc/pull/1068).

Requires https://github.com/matrix-org/matrix-doc/pull/1068 (and includes one line of it where unavoidable)

### CODE REVIEW

File: api/identity/lookup.yaml [hunk 1]
```diff
@@ -82,4 +82,57 @@ paths:
...
+            properties:
+              threepids:
+                type: array
+                items:
+                  type: array
+                  title: 3PID mappings
+                  items:
+                    type: string
+                    title: 3PID medium or address
+                description: an array of arrays containing the `3PID Medium type`_ in first position and the 3PID Address in second position.
```
Comment (richvdh) [inline api/identity/lookup.yaml]: broken link I think

File: api/identity/lookup.yaml [hunk 2]
```diff
@@ -82,4 +82,57 @@ paths:
                 description: The unix timestamp at which the association was verified.
               signatures:
                 type: object
-                description: The signatures of the verifying identity service which show that the association should be trusted, if you trust the verifying identity service.
+                description: The signatures of the verifying identity services which show that the association should be trusted, if you trust the verifying identity services.
+  "/bulk_lookup":
```
Comment (richvdh) [inline api/identity/lookup.yaml]: is this rate-limited in some way to stop people searching the whole of the misisdn space? Should we document it if so?

File: api/identity/lookup.yaml (hunk 1 above)
Comment (richvdh) [reply api/identity/lookup.yaml]: incidentally, what a horrible API...

File: api/identity/lookup.yaml [hunk 3]
```diff
@@ -82,4 +82,57 @@ paths:
...
+            properties:
+              threepids:
+                type: array
+                items:
+                  type: array
+                  title: 3PID mappings
+                  items:
+                    type: string
+                    title: 3PID medium or address or the Matrix ID
+                description: an array of array containing the `3PID Medium type`_ in first position, the 3PID Address in second position and Matrix ID is in third position.
```
Comment (richvdh) [inline api/identity/lookup.yaml]: s/is in/in/

File: api/identity/lookup.yaml (hunk 2 above)
Comment (richvdh) [reply api/identity/lookup.yaml]: feel free to leave it in the interests of just landing something

Comment (dbkr) [reply api/identity/lookup.yaml]: It isn't

**Pull request merged**